feat(config): `prOptions` (#31381)
renovatebot/renovate: docs/usage/configuration-options.md

!!! note
    If the option `minimumReleaseAge` is non-zero then Renovate disables the `prNotPendingHours` functionality.

## prOptions

Use `prOptions` to set pr-specific configuration options provided by Renovate.

Example:

```json
{
  "prOptions": {
    "gitLabIgnoreApprovals": true
  }
}
```

> You can use these options:

Reviewer: ```suggestion
You can use these configuration options within `prOptions`:

- `azureWorkItemId`
- `bbUseDefaultReviewers`
- `gitLabIgnoreApprovals`
```

Explain that we mean you can use these options within `prOptions`, and list the options.

Even better would be to autogenerate a table of allowed options within `prOptions`. That way, when we later add new options, the docs are updated automatically. I think we already do this for other bits of the docs.

### azureWorkItemId

When creating a PR in Azure DevOps, some branches can be protected with branch policies to [check for linked work items](https://docs.microsoft.com/en-us/azure/devops/repos/git/branch-policies?view=azure-devops#check-for-linked-work-items).

> Creating a work item in Azure DevOps is beyond the scope of Renovate, but Renovate can link an already existing work item when creating PRs.

Reviewer: ```suggestion
Renovate will not create a work item in Azure DevOps, this is beyond the scope of Renovate.
But Renovate can _link_ to an existing work item when Renovate creates a PR.
```

Split up sentence into two parts.

Can you configure a `azureWorkItemId` and then forget about it, as it just works? Or do you need to update the `azureWorkItemId` later?

### bbUseDefaultReviewers

> Configuring this to `true` means that Renovate will detect and apply the default reviewers rules to PRs (Bitbucket only).

Reviewer: ```suggestion
If `bbUseDefaultReviewers` is set to `true`, Renovate detects and applies the default reviewers rules to its PRs (Bitbucket only).
```

### gitLabIgnoreApprovals

> Ignore the default project level approval(s), so that Renovate bot can automerge its merge requests, without needing approval(s).

Reviewer: ```suggestion
Ignore the default project level approval(s), to allow Renovate to automerge its Merge Requests, without needing approval(s).
```

> Under the hood, it creates a MR-level approval rule where `approvals_required` is set to `0`.

Reviewer: ```suggestion
Under the hood, it creates a MR-level approval rule, where `approvals_required` is set to `0`.
```

> This option works only when `automerge=true` and either `automergeType=pr` or `automergeType=branch`.

Reviewer: ```suggestion
For `gitLabIgnoreApprovals` to work you must set `automerge=true`, and either set `automergeType=pr` _or_ set `automergeType=branch`.
```

> Also, approval rules overriding should not be [prevented in GitLab settings](https://docs.gitlab.com/ee/user/project/merge_requests/approvals/settings.html#prevent-editing-approval-rules-in-merge-requests).

Reviewer: ```suggestion
Also, approval rules overriding should _not_ be [prevented in GitLab settings](https://docs.gitlab.com/ee/user/project/merge_requests/approvals/settings.html#prevent-editing-approval-rules-in-merge-requests).
```
